packages/flutter/lib/src/foundation/serialization.dart

```diff
@@ -13,10 +13,12 @@ import 'dart:typed_data';
 /// The byte order used is [Endian.host] throughout.
 class WriteBuffer {
   /// Creates an interface for incrementally building a [ByteData] instance.
-  factory WriteBuffer() {
+  /// [startCapacity] determines the start size of the [WriteBuffer].  The closer
```

Comment:
nit: initial capacity in bytes. Also assert for  >= 0

Reply:
done

Reply:
I would assert >= 0, `0` is still a valid initial buffer size